src/charm.py

```diff
@@ -1153,32 +1189,31 @@ def set_secret(self, scope: str, key: str, value: Optional[str]) -> Optional[str
         if not value:
             return self.remove_secret(scope, key)
 
-        label = generate_secret_label(self, scope)
-        secret = self.secrets.get(label)
-        if not secret:
-            self.secrets.add(label, {key: value}, scope)
+        peers = self.model.get_relation(Config.Relations.PEERS)
+
+        if not peers:
+            return
+
+        label = None
+        if scope == APP_SCOPE:
+            self.peer_relation_app.update_relation_data(peers.id, {key: value})
```

Comment:
both `update_relation_data([...])` with `set_secret(peer.id, key,value)` - like in [postgres charm](https://github.com/canonical/postgresql-operator/blob/eef3555a17c315cfda329abd12eeb22401ed7b90/src/charm.py#L319)